### ISSUE TITLE
glReadPixels returns 0 - Need help

### ISSUE THREAD

**Issue by RoopanJK** [pthom/hello_imgui#127]
Hello, I have a application that needs to use the glReadPixels to read the pixels data of a child window and store them into a image using stb_image_write. But my issue is glReadPixels always returns 0. I have not created any additional frame buffers. So only the default frame buffer exists. I have also tried creating a custom framebuffer and binding it while drawing and reading but it also returns 0.

I have added a snapshot of how I use the glReadPixel function (just to test).
![image](https://github.com/user-attachments/assets/01f84394-ed85-4ff3-8779-bd01ac64b9a4)

I've tested it with a dear ImGui example and glReadPixel returns the pixel value. I don't know what I am doing wrong here. Any advice would be helpful. Thanks in advance.

OpenGL Version: 4.6
OS: Ubuntu 20.04

**Comment by pthom** [comment]
Hello,

Next time, try provide a minimum reproducible example with full source code _as text_.

You may take inspiration from this:

https://github.com/pthom/hello_imgui/blob/52ecf4595d7283761c9ab0c3702fa35058f2c10e/src/hello_imgui/internal/backend_impls/opengl_setup_helper/opengl_screenshot.cpp#L14-L61

**Comment by RoopanJK** [comment]
Hi there pthom. Sorry for the inconvenience, I shall provide reproducible code from next time onward.

So I figured out that I was calling the glReadPixels function before ImGui::render() call, which would eventually result in just it reading 0's. After looking into the source code I found that I should add the glReadPixles in the AfterSwap callback.

https://github.com/pthom/hello_imgui/blob/52ecf4595d7283761c9ab0c3702fa35058f2c10e/src/hello_imgui/internal/backend_impls/abstract_runner.cpp#L1371-L1384

Once I added the glReadPixels in the "callbacks.AfterSwap" in runner params. I was able to successfully read the pixels and export it as a png image using stb image library.

Many thanks pthom for a swift reply. Cheers.
